lib/brakeman.rb

```diff
@@ -498,6 +502,19 @@ def self.load_brakeman_dependency name, allow_fail = false
     end
   end
 
+  # Returns false if the specified ignore file exists and contains an ignored
+  # warning without a note.
+  # Returns true otherwise.
+  def self.all_ignore_file_entries_have_notes? file
```

Comment:
If this logic were moved to `Brakeman::Commandline` then the error message could list which warnings (fingerprints?) are missing notes.

This method could remain the same except for the call to `empty?`.

Reply:
New sample error message:
```
Error: notes required for all ignored warnings when --ensure-ignore-notes is set. No notes provided for these alerts: ["734d17c982165a84df11282ba1f87ead6017df7be9d2b44c816dfae0a51fb8ac"]
```